datafusion/common/src/tree_node.rs

```diff
@@ -33,27 +46,17 @@ use crate::Result;
 /// [`PhysicalExpr`]: https://docs.rs/datafusion/latest/datafusion/physical_plan/trait.PhysicalExpr.html
 /// [`LogicalPlan`]: https://docs.rs/datafusion-expr/latest/datafusion_expr/logical_plan/enum.LogicalPlan.html
 /// [`Expr`]: https://docs.rs/datafusion-expr/latest/datafusion_expr/expr/enum.Expr.html
-pub trait TreeNode: Sized + Clone {
-    /// Returns all children of the TreeNode
-    fn children_nodes(&self) -> Vec<Cow<Self>>;
-
+pub trait TreeNode: Sized {
     /// Use preorder to iterate the node on the tree so that we can
     /// stop fast for some cases.
     ///
     /// The `op` closure can be used to collect some info from the
     /// tree node or do some checking for the tree node.
```

Comment:
Perhaps this documentation can be updated, something like

```suggestion
    /// Applies `op` to the node and its children, with traversal controlled by 
    /// [`VisitRecursion`]
    ///
    /// The `op` closure can be used to collect some info from the
    /// tree node or do some checking for the tree node.
```